frontend/src/store/actions/actions.js

```diff
@@ -0,0 +1,17 @@
+import * as actionTypes from "../actionTypes";
+import { fetchData } from "../../utils";
+
+const saveCategories = categories => {
+  return {
+    type: actionTypes.GET_ALL_CATEGORIES,
+    categories
+  };
+};
+
+export const getCategories = () => {
+  return async dispatch => {
+    const categoriesArray = await fetchData("/api/categories");
+
+    dispatch(saveCategories(categoriesArray));
+  };
+};
```

Comment:
A bit confusing, saveCartegories using GET_ALL_CATEGORIES.
Change names in these two methods.
Actions name, should be based on type name.

```suggestion
const getAllCategories = categories => {
  return {
    type: actionTypes.GET_ALL_CATEGORIES,
    categories
  };
};

export const saveAllCategories = () => {
  return async dispatch => {
    const categoriesArray = await fetchData("/api/categories");

    dispatch(getAllCategories(categoriesArray));
  };
};
```